ui/src/tempVars/components/TemplateVariableEditor.tsx

```diff
@@ -373,6 +416,22 @@ class TemplateVariableEditor extends PureComponent<Props, State> {
     return 'Save'
   }
 
+  // When we change the source database here and below, we want to reset the template complete
+  private handleSelectDynamicSource = () => {
+    this.setState({
+      isDynamicSourceSelected: true,
```

Comment:
Why not a `type` here instead of a boolean?  Something like `type SelectedSourceType = 'static' | 'dynamic'`?  Then `this.setState({sourceType: 'dynamic'})` etc. Just a style suggestion. Functionally it will be the same. Don't particularly care for booleans.  Limits us to only two cases.  I think this would read nicely:

```
if (sourceType === 'dynamic') {
 ... do dynamic stuff here
}
```

Reply:
We talked about this offline a little bit - I'm going to hold off on doing this right now.

I agree with the assessment 100% The issue is that this pattern is used elsewhere and even though it's a little wonky, I think having consistent wonk for a short while is less confusing than having some wonk and some Done Right™ implementations.

We're planning on having a follow up 1.8.1 release, this might be worth thinking about for that.